File: PYTHON/Pertemuan 01/MahasiswaAkademik.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
class MahasiswaAkademik:
    def __init__(self, nama: str, nim: str, ipk: float, semester: int, prodi: str):
        self.nama = nama
        self.nim = nim
        self.ipk = ipk
        self.semester = semester
        self.program_studi = prodi

    def ambil_mata_kuliah(self, mk: str):
        print(f"{self.nama} mengambil mata kuliah: {mk}")

    def hitung_ipk(self, nilai_list: list[float]):
        if nilai_list:
            self.ipk = sum(nilai_list) / len(nilai_list)
        print(f"{self.nama} – IPK diperbarui menjadi: {self.ipk:.2f}")

    def tampilkan_info(self):
        print("  ┌─ Kartu Akademik ──────────────────────────┐")
        print(f"  │ Nama         : {self.nama}")
        print(f"  │ NIM          : {self.nim}")
        print(f"  │ IPK          : {self.ipk:.2f}")
        print(f"  │ Semester     : {self.semester}")
        print(f"  │ Program Studi: {self.program_studi}")
        print("  └───────────────────────────────────────────┘")

File: JAVA/Pertemuan 01/src/id/ac/uph/pbo/pertemuan01/MahasiswaAkademik.java
package id.ac.uph.pbo.pertemuan01;

public class MahasiswaAkademik {

    private String nama;
    private String nim;
    private double ipk;
    private int    semester;
    private String programStudi;

    public MahasiswaAkademik(String nama, String nim,
                              double ipk, int semester, String prodi) {
        this.nama         = nama;
        this.nim          = nim;
        this.ipk          = ipk;
        this.semester     = semester;
        this.programStudi = prodi;
    }

    // Proses abstraksi: hanya behaviour yang relevan untuk sistem akademik
    public void ambilMataKuliah(String mk) {
        System.out.println(nama + " mengambil mata kuliah: " + mk);
    }

    public void hitungIPK(double[] nilaiList) {
        double total = 0;
        for (double n : nilaiList) total += n;
        ipk = total / nilaiList.length;
        System.out.printf("%s – IPK diperbarui menjadi: %.2f%n", nama, ipk);
    }

    public void tampilkanInfo() {
        System.out.println("  ┌─ Kartu Akademik ──────────────────────────┐");
        System.out.println("  │ Nama         : " + nama);
        System.out.println("  │ NIM          : " + nim);
        System.out.printf ("  │ IPK          : %.2f%n", ipk);
        System.out.println("  │ Semester     : " + semester);
        System.out.println("  │ Program Studi: " + programStudi);
        System.out.println("  └───────────────────────────────────────────┘");
    }
}
